frontend/src/components/BalanceCards.jsx

```diff
@@ -6,32 +6,56 @@ import { ReactComponent as DAI } from '../assets/icons/dai.svg';
 import { getBalances } from '../utils/wallet';
 
 const BalanceCards = ({ walletId }) => {
-    const [balances, setBalances] = useState([
-        { icon: <ETH />, title: 'ETH', balance: '0.00' },
-        { icon: <USDC />, title: 'USDC', balance: '0.00' },
-        { icon: <STRK />, title: 'STRK', balance: '0.00' },
-        { icon: <DAI />, title: 'DAI', balance: '0.00' },
-    ]);
+  const [balances, setBalances] = useState([
+    { icon: <ETH />, title: 'ETH', balance: '0.00' },
+    { icon: <USDC />, title: 'USDC', balance: '0.00' },
+    { icon: <STRK />, title: 'STRK', balance: '0.00' },
+    { icon: <DAI />, title: 'DAI', balance: '0.00' },
+  ]);
 
-    useEffect(() => {
-        getBalances(walletId, setBalances);
-    }, [walletId]);
+  const [isMobile, setIsMobile] = useState(window.matchMedia('(max-width: 768px)').matches);
+  useEffect(() => {
+    getBalances(walletId, setBalances);
+  }, [walletId]);
 
+  useEffect(() => {
+    const mediaQuery = window.matchMedia('(max-width: 768px)');
 
-    return (
-        <div className="balance-container">
-            {balances.map((balance) => (
-            <div className={"balance-item"} key={balance.title}>
-                <label htmlFor={balance.title} className={"balance-title"}>
-                {balance.icon}{balance.title} Balance:
-                </label>
-                <label htmlFor={balance.title}>
-                {balance.balance}
-                </label>
+    const handleMediaQueryChange = (e) => {
+      setIsMobile(e.matches);
+    };
+
+    mediaQuery.addEventListener('change', handleMediaQueryChange);
+
+    return () => {
+      mediaQuery.removeEventListener('change', handleMediaQueryChange);
+    };
+  }, []);
```

Comment:
Please create separate custom hook `useMatchMedia` and use it instead

Reply:
Alright